doc/SONiC_QoS_Mgmt_HLD-1.0.md

```diff
@@ -0,0 +1,499 @@
+# QoS
+
+# High Level Design Document
+#### Rev 1.0
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 1.0 | 10/15/2019  | Oliver Hu, Ashok Daparthi, Srinadh Penugondaa, Eddy Lem | Initial version  |
+
+
+# About this Manual
+This document provides general information about QoS configuration in SONiC using the management framework.
+
+# Scope
+Covers Northbound interface for the QoS feature, as well as Unit Test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|:-------------------------|:------------------------------------|
+| QoS                      | Quality of Service                 |
+
+# 1 Feature Overview
+
+Provide management framework capabilities to handle:
+- Traffic classification
+- Queue management
+- Traffic scheduling
+- Buffer Threshold configuration
+- Watermark statistics
+- Associated show commands.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide management framework support to existing SONiC capabilities with respect to QoS.
+
+### 1.1.2 Configuration and Management Requirements
+- CLI configuration and show commands
+- REST API support
+- gNMI Support
+
+Details described in Section 3.
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+QoS deals with the traffic prioritization and scheduling. 
+
+The basic approach is to classify different traffic to different service category, ie. putting traffic to different queues during ingress packet processing. 
+
+In addition defining different queue behavior controls how the traffic is serviced during egress packet processing.   
+
+### 1.2.2 Container
+Management container
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+Anywhere user wants to give different traffic with different treatment.
+
+## 2.2 Functional Description
+
+Functionally QoS provides user the capability to handle different traffic with different level of service, from separate queuing to different scheduling behavior.  
+
+# 3 Design
+## 3.1 Overview
+
+To support quality of service, first user has to have a way to differentiate different types of traffic. Provided here is the method based on the DSCP value of the packet. Packets are classified into different internal traffic class based on their DSCP value and will be sent to different queues for scheduling. 

```

Comment:
What is the default dscp-traffic-class maping?
How packets will be classified when the L2 switched packets (or packets without dscp value, may be negative scenario)